src/NServiceBus.NHibernate/CallbackList.cs

```diff
@@ -0,0 +1,48 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    class CallbackList
+    {
+        Func<Task> singleItem;
+        List<Func<Task>> multipleItems;
+
+        public void Add(Func<Task> callback)
```

Comment:
Is the effect of storing single item separately measurable? It would be good if this complexity was justified by a significant benefit.